tests/src/Interop/StringMarshalling/Native/StringMarshalingNative.h

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <platformdefines.h>
+#include <type_traits>
+#include <algorithm>
+
+template<typename StringT>
+struct StringTraits
+{
+    using type = StringT;
+    using CharT = typename std::remove_pointer<StringT>::type;
+    using ConstStringT = CharT const *;
+    static const bool VerifyNullTerminator = true;
+};
+
+struct BSTRTraits
```

Comment:
Can't we specialize a version of `struct StringTraits<>`?

Reply:
Sadly since `BSTR` is technically the same type as `LPWSTR` to the type system we can't.

Reply:
Good point. I actually remember hitting that issue as well on a previous project. Okay.